File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.go
func maximumHappinessSum(happiness []int, k int) int64 {
    sort.Slice(happiness, func(i, j int) bool {
        return happiness[i] > happiness[j]
    })
    ans := int64(0)
    for i := 0; i < k; i++ {
        ans += max(0, int64(happiness[i] - i))
    }
    return ans
}

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.cpp
#pragma GCC optimize("03", "unroll-loops");
static const int __ = [](){
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    return 0;
}();
using ll = long long;
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.begin(), happiness.end(), greater<>());
        ll ans = 0;
        for(int i = 0; i < k; i++) {
            ans += max(0, happiness[i] - i);
        }

        return ans;
    }
};